src/main/scala/com/thenewmotion/akka/rabbitmq/ConnectionActor.scala

```diff
@@ -124,7 +124,7 @@ class ConnectionActor(
     log.debug("{} closing broken connection {}", self.path, connection)
     closeIfOpen(connection)
 
-    self ! Connect
+    setTimer(reconnectTimer, Connect, reconnectionDelay / 3, repeat = false)
```

Comment:
And I still don't see how it is a functional bug. The ChannelActor always processes each ChannelMessage with a valid channel that will *not* be closed while the operation is performed on the channel.

There is a performance penalty for renewing the channel twice instead of once of course, but indeed, I see no way to avoid that altogether.